test/strscan/test_stringscanner.rb

```diff
@@ -1128,6 +1128,15 @@ def test_fixed_anchor_false
     assert_equal(false, StringScanner.new("a", fixed_anchor: nil).fixed_anchor?)
     assert_equal(false, StringScanner.new("a", fixed_anchor: false).fixed_anchor?)
   end
+
+  def test_large_string
+    six_gib = (6 * 1024 * 1024 * 1024)
+    string = "a" * six_gib
+    s = create_string_scanner(string) # 6 GiB string
```

Comment:
Does this mean that we need at least 6 GiB memory to run this test...?

Reply:
Indeed. I don't know how to test it otherwise. I could reduce the size to a bit more than 4GiB, or even 2GiB given it's signed.

We could also just not test it, but this revealed that the Java version has the same limitation, so I think it was valuable. (I won't update the JRuby version though, because there everything uses an int so it's way to much work for me. cc @headius)

Reply:
I ended up removing the test, as it also doesn't work on windows, nor on 32bit systems.

Reply:
OK.